File: postkit/com.wapo.rainbow.article.model/src/main/java/com/wapo/flagship/json/AdConfig.java
package com.wapo.flagship.json;

import java.io.Serializable;

/**
 * @author davisas3 on 3/17/15.
 */
public class AdConfig implements Serializable {

    private String adSetCode;
    private String commercialAdNode;
    private boolean playAds;
    private boolean enableAutoPreview;
    private boolean forceAd;
    private boolean playVideoAds;
    private String adZone;
    private boolean autoPlayPreroll;
    private AdSetUrls adSetUrls;
    private AdSetConfig adSetConfig;

    public String getAdSetCode() {
        return adSetCode;
    }

    public void setAdSetCode(String adSetCode) {
        this.adSetCode = adSetCode;
    }

    public String getCommercialAdNode() {
        return commercialAdNode;
    }

    public void setCommercialAdNode(String commercialAdNode) {
        this.commercialAdNode = commercialAdNode;
    }

    public boolean isPlayAds() {
        return playAds;
    }

    public void setPlayAds(boolean playAds) {
        this.playAds = playAds;
    }

    public boolean isEnableAutoPreview() {
        return enableAutoPreview;
    }

    public void setEnableAutoPreview(boolean enableAutoPreview) {
        this.enableAutoPreview = enableAutoPreview;
    }

    public boolean isForceAd() {
        return forceAd;
    }

    public void setForceAd(boolean forceAd) {
        this.forceAd = forceAd;
    }

    public boolean isPlayVideoAds() {
        return playVideoAds;
    }

    public void setPlayVideoAds(boolean playVideoAds) {
        this.playVideoAds = playVideoAds;
    }

    public String getAdZone() {
        return adZone;
    }

    public void setAdZone(String adZone) {
        this.adZone = adZone;
    }

    public boolean isAutoPlayPreroll() {
        return autoPlayPreroll;
    }

    public void setAutoPlayPreroll(boolean autoPlayPreroll) {
        this.autoPlayPreroll = autoPlayPreroll;
    }

    public AdSetUrls getAdSetUrls() {
        return adSetUrls;
    }

    public AdSetConfig getAdSetConfig() {
        return adSetConfig;
    }
}

File: android-tablet/src/main/java/com/wapo/flagship/features/articles2/models/deserialized/video/Adconfig.kt
package com.wapo.flagship.features.articles2.models.deserialized.video

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Adconfig(
    @Json(name = "adSetConfig")
    val adSetConfig: AdSetConfig?,
    @Json(name = "adSetUrls")
    val adSetUrls: AdSetUrlsX?,
    @Json(name = "allowPrerollOnDomain")
    val allowPrerollOnDomain: Boolean?,
    @Json(name = "autoPlayPreroll")
    val autoPlayPreroll: Boolean?,
    @Json(name = "commercialAdNode")
    val commercialAdNode: String?,
    @Json(name = "enableAdInsertion")
    val enableAdInsertion: Boolean?,
    @Json(name = "enableAutoPreview")
    val enableAutoPreview: Boolean?,
    @Json(name = "enableServerSideFallback")
    val enableServerSideFallback: Boolean?,
    @Json(name = "forceAd")
    val forceAd: Boolean?,
    @Json(name = "playAds")
    val playAds: Boolean?,
    @Json(name = "playVideoAds")
    val playVideoAds: Boolean?,
    @Json(name = "videoAdZone")
    val videoAdZone: String?,
    @Json(name = "primarySectionID")
    val primarySectionId: String?
)
